CMakeLists.txt

```diff
@@ -88,44 +87,36 @@ list(PREPEND CMAKE_PREFIX_PATH ${CMAKE_CURRENT_BINARY_DIR}/faabric)
 include(${CMAKE_CURRENT_BINARY_DIR}/faabric/conan_paths.cmake)
 find_package(cpprestsdk REQUIRED)
 
-# Detect SGX
+# ----------------------------------------
+# SGX configuration
+# ----------------------------------------
+
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
-if (FAASM_SGX)
+if (NOT (FAASM_SGX_MODE STREQUAL "Disabled"))
+    # Set maro definitions for conditional compilation
+    if (FAASM_SGX_MODE STREQUAL "Hardware")
+        add_definitions(-DFAASM_SGX_HARDWARE_MODE)
+    elseif (FAASM_SGX_MODE STREQUAL "Simulation")
+        add_definitions(-DFAASM_SGX_SIMULATION_MODE)
+    else ()
+        message(FATAL_ERROR "Unrecognised SGX mode: ${FAASM_SGX_MODE}")
+    endif ()
```

Comment:
AFAICT the variable has a default value, through the `option` command, to `Simulation`. I could change it to `Disabled` if you prefer, but this would prevent the SGX tests from running (by default).

Reply:
Ah ok yes, sorry I was getting confused with this issue we saw before: https://github.com/faasm/faasm/pull/596. Will we see a similar problem here?

Reply:
I have followed Jakub's advice and always set the variable through the Python tasks, instead of relying on CMake's variable caching.